src/metrics/mongoose_metrics_api.erl

```diff
@@ -16,30 +16,59 @@
     {ok, #{binary() => binary() | [metric_dict_result()]}}
     | {error, binary()}.
 
--spec get_metrics(Name :: name()) -> {ok, [metric_result()]}.
+-spec get_metrics(Name :: name()) -> {ok, [metric_result()]} | {no_name_found, binary()}.
 get_metrics(Name) ->
-    Values = exometer:get_values(Name),
-    {ok, lists:map(fun make_metric_result/1, Values)}.
+    case exometer:get_values(Name) of
+        [] ->
+            {no_name_found, <<"No value found for queried name">>};
+        Values ->
+            {ok, lists:map(fun make_metric_result/1, Values)}
+    end.
 
 -spec get_metrics_as_dicts(Name :: name(), Keys :: [key()]) ->
-    {ok, [metric_dict_result()]}.
+    {ok, [metric_dict_result()]} | {no_name_found | no_key_found, binary()}.
 get_metrics_as_dicts(Name, Keys) ->
-    Values = exometer:get_values(Name),
-    {ok, [make_metric_dict_result(V, Keys) || V <- Values]}.
+    case exometer:get_values(Name) of
+        [] ->
+            {no_name_found, <<"No value found for queried names">>};
+        Values ->
+            Results = [make_metric_dict_result({N, Dict}, Keys) ||
+                       {N, Dict} <- Values, filter_keys(Dict, Keys) =/= []],
+            case Results of
+                [] ->
+                    {no_key_found, <<"No value found for queried keys">>};
```

Comment:
"No matching metrics.".

No value is usually means the metric exists but for some reason it is null.
Though an empty list of matches is already `No matching metrics`.

But I think the old behaviour of matching exometer API was fine.
Because how would we use API usually?
- Query for a list of metrics that match prefix, like `global`.
- Find the metric we are actually interested in.
- If it does not exist for some reason, we can handle it on the client side still.

Here we just adding one extra format for results (i.e. results with errors), which just making API harder to use.
Because currently it does not add any new information, useful for the user.

My opinion though.

And for keys: keys are just for filtering the results. It is an extra feature, behaviour ideally should be the same as when we don't provide keys. Also, we cannot really give a meaningless error when one key has a typo (`onw` for example). OK, we can, but we should define all possible keys somewhere in this case. But again, should we be that strict? Because it is check for one filter key, other filter keys still can work fine. It could cause problems when user switches between versions.